Plans redesign: Available and missing features

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

This PR implements the experiment that redesigns the signup flow plans step. Users in the treatment group will see the new feature comparison section for each plan in the signup plans step.

NOTE: 
1. This PR uses the changes in https://github.com/Automattic/wp-calypso/pull/50005 as the base branch and is built on top of it.

**Important links**
1. [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813)
2. pau2Xa-2yE-p2 for the project thread.


**Annual plans view with free subdomain selected (the first feature item is a generic free domain message)**

![calypso localhost_3000_start_plans (5)](https://user-images.githubusercontent.com/1269602/108342053-5cf41a80-7200-11eb-808a-974d928bc6db.png)

**Annual plans view with a custom domain selected (the first feature item mentions the selected domain)**

![calypso localhost_3000_start_plans (7)](https://user-images.githubusercontent.com/1269602/108342101-6d0bfa00-7200-11eb-9d03-656a5f0fbc23.png)

**Monthly plans view with a free subdomain selected (Live chat and free domain mentioned as unavailable)**

![calypso localhost_3000_start_plans (6)](https://user-images.githubusercontent.com/1269602/108342132-74cb9e80-7200-11eb-88ce-707857c54e43.png)


**Monthly plans view with a custom domain selected (Live chat and free domain mentioned as unavailable)**

![calypso localhost_3000_start_plans (8)](https://user-images.githubusercontent.com/1269602/108342156-7bf2ac80-7200-11eb-9264-72de3866f61a.png)


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open http://calypso.localhost:3000/ and assign yourself to treatment of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Begin signup at `/start` and verify that the plans step UI matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813). Verify for both free subdomain and custom domain selections.
* Verify that the UI matches the Figma design for both annual and monthly plan views.
* Confirm that in Calypso's /plans view, the UI is unchanged.
* Also confirm that the launch flow UI is unchanged.
* Verify that control experience is unchanged.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/pull/49877

niranjan-uma-shankar committed:
Show available and missing features in a side-by-side layout

a8ci18n commented on client/lib/plans/features-list.js:
```diff
@@ -253,6 +259,12 @@ export const FEATURES_LIST = {
 
+	[ constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR ]: {
+		getSlug: () => constants.FEATURE_ADVANCED_SEO_EXPANDED_ABBR,
+		getTitle: () => i18n.translate( 'Advanced SEO (Search Engine Optimisation) tools' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Advanced SEO (Search Engine Optimization) tools' )` *ES Score: 6*

a8ci18n commented on client/lib/plans/features-list.js:
```diff
@@ -542,6 +560,14 @@ export const FEATURES_LIST = {
+		getSlug: () => constants.FEATURE_INSTALL_PLUGINS,
+		getTitle: () =>
+			i18n.translate(
+				'Access to more than 50,000 WordPress plugins to extend functionality for your site'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Install plugins and extend functionality for your site with access to more than 50,000 WordPress plugins' )` *ES Score: 8*

niranjan-uma-shankar committed:
Show notice that certain monthly features are only available with annual plans

niranjan-uma-shankar committed:
Fix merge conflict

niranjan-uma-shankar committed:
Fix merge conflicts

niranjan-uma-shankar committed:
Remove commented out lines, and remove an unnecessary if condition

niranjan-uma-shankar committed:
Force prettier

ollierozdarz commented:
Great job with these, @niranjan-uma-shankar! 🙂 

I've got a few quick comments below, mainly as a FYI. 

**Billing toggle**

@olaolusoga recently created a really nice [billing toggle](https://www.figma.com/proto/NhFdwnznk6LEuVtvc18qRm/LOHP-pricing-test?node-id=4%3A6&viewport=471%2C378%2C0.4404674172401428&scaling=min-zoom) for the Control pricing pages (via the Payments team). It might be worth incorporating that as I'm sure it will increase the usability.

<img width="1092" alt="Screen Shot 2021-02-16 at 1 58 17 pm" src="https://user-images.githubusercontent.com/67085494/108016878-43679d00-705f-11eb-9f55-73d5dcd2d379.png">

**Unavailable features**

Because we're showing available and unavailable features in the comparison table below each plan, it could be worth using a more subdued Grey (or opacity) for the unavailable features so it's easier to quickly see the value.

<img width="668" alt="Screen Shot 2021-02-16 at 2 04 42 pm" src="https://user-images.githubusercontent.com/67085494/108017226-149df680-7060-11eb-930c-52af44f677a5.png">

**'Included with..' labels**

For the 'Included with..' labels, I believe Luna have included these above the strikeout text in New onboarding. It might be worth doing the same so we're keeping consistency. Also, in the same vein, we could use the same font-weight for both labels (the heavier weight is preferable).

<img width="433" alt="Screen Shot 2021-02-16 at 2 06 31 pm" src="https://user-images.githubusercontent.com/67085494/108017375-72324300-7060-11eb-9242-b98d6771f63c.png">

**Domain is free for first year**

Lastly, you may have a better idea of what converts better here, but '_The domain example.blog is free for the first year_' feels like it could be shortened to '_example.blog is free for the first year_'. We also say 'Free for One Year', and it might be good to find some consistency in the wording. 
 
Please let me know how else I can help!

niranjan-uma-shankar committed:
Make the unavailable features in a light shade of gray; use a different version of unlimited premium themes feature

niranjan-uma-shankar committed:
Design feedback: Move the "included in annual plans" copy above the feature

niranjan-uma-shankar commented:
Thanks for the feedack @ollierozdarz . 

> Billing toggle

Thanks, I'll implement that.

> 'Included with..' labels

I've implemented the suggested changes and updated the screenshots on the PR description.


> Lastly, you may have a better idea of what converts better here, but 'The domain example.blog is free for the first year' feels like it could be shortened to 'example.blog is free for the first year'. 

I'll check with Marketing folks on this but the reason why I went for a longer form "The domain ..." is that we've recently been getting reports of users confusing a free domain to mean that the plan itself is free(related issue in https://github.com/Automattic/dotcom-manage/issues/408).

> Because we're showing available and unavailable features in the comparison table below each plan, it could be worth using a more subdued Grey (or opacity) for the unavailable features so it's easier to quickly see the value.

Great point. I've pushed in a change to fix this and now the page looks like this:

<img width="1284" alt="Screenshot 2021-02-16 at 7 36 47 PM" src="https://user-images.githubusercontent.com/1269602/108073683-547bd280-708e-11eb-9ef5-796bc70b114c.png">


I'd also like to mention here a few points raised by @jaswrks in Slack, and I'd love to hear your take on them. I'm paraphrasing his comments:

* @jaswrks mentions that it's not immediately obvious that features are cascading horizontally and that you get more going from left to right. Could a table grid be helpful here(similar to the large comparison table in wordpress.com/pricing)?
* @jaswrks mentions that the features appear like a wall of text. Either we could increase the contrast between available and missing features(update: this is already done) or we could entirely remove the unavailable features(which is how it's shown in the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1457%3A5140)).

@ollierozdarz What do you think? My personal take is - (1) I don't have strong opinions here, so I defer to you (2) I think the subdued grey has really helped with the readability and looks okay to me but I'm not opposed to entirely removing the unavailable X marked features if it can improve readability.

ollierozdarz commented:
> What do you think?  My personal take is - (1) I don't have strong opinions here, so I defer to you (2) I think the subdued grey has really helped with the readability and looks okay to me but I'm not opposed to entirely removing the unavailable X marked features if it can improve readability.

@niranjan-uma-shankar I'm **+1** for removing the unavailable features. I haven't included the unavailable features in my newest iteration of the Reskinned onboarding because I personally felt like there was too much to take in at once. Also, by removing those (now subdued) unavailable features, it may help @jaswrks feedback on the left to right parsing. 

**Example:**
<img width="1182" alt="Screen Shot 2021-02-17 at 2 04 15 pm" src="https://user-images.githubusercontent.com/67085494/108154683-0ff14500-7129-11eb-8673-bba860b54f78.png">

jaswrks commented:
Thanks for considering my feedback. Really appreciate. Design example in last screenshot looks amazing. I like the toggle above it as well.

dzver commented:
When I select a paid domain, the plans page works and looks like a screenshot.

Selecting the free domain, however, leads to a JS error.

![signup-plans-js-err](https://user-images.githubusercontent.com/82778/108175390-62bb0480-7109-11eb-98ef-f6d8bf6f35c4.gif)

niranjan-uma-shankar committed:
Remove missing features from the UI

niranjan-uma-shankar committed:
Fix bug that shows a JS error if a free subdomain is selected

niranjan-uma-shankar committed:
Fix included in annual plans notice

niranjan-uma-shankar committed:
Mention live chat support as 24x7 for business and ecommerce plans

niranjan-uma-shankar committed:
Merge branch 'add/action-button-new-plans-design-signup' into add/new-plan-step-compare-features

niranjan-uma-shankar committed:
Add some padding for the last feature in the last column

niranjan-uma-shankar committed:
Merge branch 'add/action-button-new-plans-design-signup' into add/new-plan-step-compare-features

niranjan-uma-shankar commented:
> Selecting the free domain, however, leads to a JS error.

I've fixed this, could you please check again?

dzver reviewed (approved):
LGTM

I verified the signup flow for a new user, for a new site, the launch-site flow, and also confirmed the js error doesn't appear.

niranjan-uma-shankar committed:
Plans redesign: Add new toggle design (#50227)

* Implement the new toggle UI

* Add a hover effect for the toggle buttons

* i18n bot recommendation

* Revert
